test/test_bdf.py

```diff
@@ -0,0 +1,256 @@
+
+
+@pytest.mark.parametrize("problem, method, expected_order", [
+    [KapsProblem(epsilon=0.9), AdaptiveBDFMethodBuilder(
+        "y", fixed_order=1, max_dt_growth=10), 1],
+    [KapsProblem(epsilon=0.9), AdaptiveBDFMethodBuilder(
+        "y", fixed_order=1, max_dt_growth=10, ndf=True), 1],
+    [KapsProblem(epsilon=0.9), AdaptiveBDFMethodBuilder(
+        "y", fixed_order=2, max_dt_growth=10, bootstrap_factor=10), 2],
+    [KapsProblem(epsilon=0.9), AdaptiveBDFMethodBuilder(
+        "y", fixed_order=2, max_dt_growth=10, bootstrap_factor=10, ndf=True), 2],
+    [KapsProblem(epsilon=0.9), AdaptiveBDFMethodBuilder(
+        "y", fixed_order=3, max_dt_growth=10, bootstrap_factor=100), 3],
+    [KapsProblem(epsilon=0.9), AdaptiveBDFMethodBuilder(
+        "y", fixed_order=3, max_dt_growth=10, bootstrap_factor=100, ndf=True), 3],
+    [KapsProblem(epsilon=0.9), AdaptiveBDFMethodBuilder(
+        "y", fixed_order=4, max_dt_growth=10, bootstrap_factor=1000), 4],
+    [KapsProblem(epsilon=0.9), AdaptiveBDFMethodBuilder(
+        "y", fixed_order=4, max_dt_growth=10, bootstrap_factor=1000, ndf=True), 4],
+    [KapsProblem(epsilon=0.9), AdaptiveBDFMethodBuilder(
+        "y", fixed_order=5, max_dt_growth=10, bootstrap_factor=20000), 5],
+    [KapsProblem(epsilon=0.9), AdaptiveBDFMethodBuilder(
+        "y", fixed_order=5, max_dt_growth=10, bootstrap_factor=20000, ndf=True), 5],
+    ])
+def test_convergence(python_method_impl, problem, method, expected_order,
+                     show_dag=False, plot_solution=False):
+    pytest.importorskip("scipy")
+
+    from functools import partial
+    dts = 2**-np.array(range(3, 8), dtype=np.float64)
+    function_map = {"<func>y": problem,
+                    "<func>solver": partial(kaps_solver, problem)}
+    from utils import check_simple_convergence
+    check_simple_convergence(method=method, method_impl=python_method_impl,
+                             expected_order=expected_order, show_dag=show_dag,
+                             plot_solution=plot_solution,
+                             function_map=function_map, dts=dts,
+                             implicit=True, problem=problem,
+                             solver_hook=solver_hook)
+
+
+# {{{ adaptive test
+
+@pytest.mark.parametrize("method", [
+    AdaptiveBDFMethodBuilder(
+        "y", rtol=1e-6),
+    AdaptiveBDFMethodBuilder(
+        "y", rtol=1e-6, ndf=True),
+    ])
+def test_adaptive_timestep(python_method_impl, method, show_dag=False,
+                           plot=False):
+    pytest.importorskip("scipy")
+    from utils import check_adaptive_timestep
+    check_adaptive_timestep(python_method_impl=python_method_impl, method=method,
+                             ss_frac=0.7, bs_frac=0.2, ss_targ=0.01,
+                             bs_targ=0.05, show_dag=show_dag, plot=plot,
+                             implicit=True, solver=newton_solver,
+                             solver_hook=solver_hook)
+    # Use "DEBUG" to trace execution
```

Comment:
How come so much of this is commented out?

Reply:
This was something I did when unifying this convergence test within a utility, and was an erroneous artifact. No longer present as of 6606465.